src/components/DataRenderFunc.js

```diff
@@ -0,0 +1,236 @@
+import React, {useState, useEffect} from "react";
+import "./DataRenderStyles.css";
+import Button from 'react-bootstrap/Button';
+import { Row } from "simple-flexbox";
+import "bootstrap/dist/css/bootstrap.min.css";
+import {
+    AreaChart,
+    Area,
+    LineChart,
+    Line,
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    Legend,
+    ResponsiveContainer,
+    LabelList
+} from "recharts";
+import NumberFormat from 'react-number-format';
+
+const dataTest = [
+    { name: 'March', Tested: 4000, Hospitalized: 2400, Dead: 2400 },
+    { name: 'April', Tested: 3000, Hospitalized: 1398, Dead: 2210 },
+    { name: 'May', Tested: 2000, Hospitalized: 9800, Dead: 2290 },
+    { name: 'June', Tested: 2780, Hospitalized: 3908, Dead: 2000 },
+    { name: 'July', Tested: 1890, Hospitalized: 4800, Dead: 2181 },
+    { name: 'August', Tested: 2390, Hospitalized: 3800, Dead: 2500 },
+    { name: 'September', Tested: 3490, Hospitalized: 4300, Dead: 2100 },
+];
+
+const DataRenderFunc =(props)=>{
+    const {stateId} = props
+    
+    const [loading, setLoading] = useState(true)
+    const [allStates, setAllStates] = useState(null)
+    const [showTesting, setShowTesting] = useState(true)
+    const [showHosp, setShowHosp] = useState(false)
+    const [showDeath, setShowDeath] = useState(false)
+     const [usaState, setUsaState] = useState(null)
+
+
+        
+    // const getData = async () => {
+    //     const url = "https://api.covidtracking.com/v1/states/current.json";
+    //     const response = await fetch(url);
+    //     const data = await response.json();
+    //     console.log(data);
+    //     return data;
+    // };
+
+    // Component fires when marker is clicked
+    useEffect(()=>{
+        const getData = async () => {
+            // const result = await axios(
+                const url = "https://api.covidtracking.com/v1/states/current.json";
+        const response = await fetch(url);
```

Comment:
What happens if there is an error?

You have not checked `response.ok` So this is section is assuming an optimistic path (always 200 response).

Reply:
I haven't used .ok before, would be great to discuss this and error handling in general on our Monday call. I put the code block below under ```const response = await fetch("https://api.covidtracking.com/v1/states/current.json");```. Would definitely helpful to go over this, and seems like a best practice I should be using regularly.

```
if (response.ok) {
            console.log("Request fetched successfully!")
            return;
        }

        if (!response.ok) {
            console.log("Request failed to fetch!")
            throw new Error(response.error);
        }
```

Reply:
I just ran it again after adding the previous code block I get the error below and nothing that is in useEffect renders:

  Line 65:8:    React Hook useEffect has a missing dependency: 'stateId'. Either include it or remove the dependency array. If 'setUsaState' needs the current value of 'stateId', you can also switch to useReducer instead of useState and read 'stateId' in the reducer  react-hooks/exhaustive-deps

But when I comment it out it renders everything in useEffect as expected. I'm going to leave that block of code in there but commented out for now, would be great if we could look at this and work to debug it together on our Monday call.